AirCasting/Models/AirBeamCellularSessionCreator.swift

```diff
@@ -0,0 +1,95 @@
+// Created by Lunar on 13/09/2021.
+//
+
+import Foundation
+import CoreLocation
+
+final class AirBeamCellularSessionCreator: SessionCreator {
+    enum AirBeamSessionCreatorError: Swift.Error {
+        case invalidCreateSessionContext(CreateSessionContext)
+    }
+    let userAuthenticationSession: UserAuthenticationSession
+    let measurementStreamStorage: MeasurementStreamStorage
+    private let createSessionService: CreateSessionAPIService
+    
+    convenience init(measurementStreamStorage: MeasurementStreamStorage, userAuthenticationSession: UserAuthenticationSession, baseUrl: BaseURLProvider) {
+        self.init(measurementStreamStorage: measurementStreamStorage,
+                  createSessionService: CreateSessionAPIService(authorisationService: userAuthenticationSession, baseUrlProvider: baseUrl),
+                  userAuthenticationSession: userAuthenticationSession)
+    }
+    
+    init(measurementStreamStorage: MeasurementStreamStorage, createSessionService: CreateSessionAPIService, userAuthenticationSession: UserAuthenticationSession) {
+        self.measurementStreamStorage = measurementStreamStorage
+        self.createSessionService = createSessionService
+        self.userAuthenticationSession = userAuthenticationSession
+    }
+    
+    func createSession(_ sessionContext: CreateSessionContext, completion: @escaping (Result<Void, Error>) -> Void) {
+        guard let sessionType = sessionContext.sessionType,
+              let sessionUUID = sessionContext.sessionUUID else {
+            assertionFailure("invalidCreateSessionContext \(sessionContext)")
+            completion(.failure(AirBeamSessionCreatorError.invalidCreateSessionContext(sessionContext)))
+            return
+        }
+        // Save data to app's database
+        let session = Session(uuid: sessionUUID,
```

Comment:
idk, maybe I'm missing it, but I think this is not needed. It won't save to database by itself (this is just a DTO struct after all), and it's not used later in the func.